src/Google.Api.Gax/Api/Gax/Expiration.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Google.Api.Gax
+{
+    /// <summary>
+    /// Expiration specified by relative timeout or absolute deadline.
+    /// </summary>
+    public class Expiration
+    {
+        /// <summary>
+        /// Create an <see cref="Expiration"/> with a relative timeout.
+        /// </summary>
+        /// <param name="timeout">The relative timeout.</param>
+        /// <returns>An <see cref="Expiration"/> with the specified relative timeout.</returns>
+        /// <remarks>
+        /// Zero or negative timeouts are valid, and will cause immediate failure of the operation being performed.
+        /// </remarks>
+        public static Expiration WithTimeout(TimeSpan timeout) => new Expiration(timeout, null);
+
+        /// <summary>
+        /// Create an <see cref="Expiration"/> with an absolute deadline.
+        /// </summary>
+        /// <param name="deadline">The absolute deadline. Should be a UTC datetime.</param>
+        /// <returns>An <see cref="Expiration"/> with the specified absolute deadline.</returns>
+        /// <remarks>
+        /// Deadlines in the past are valid, and will cause immediate failure of the operation being performed.
+        /// </remarks>
+        public static Expiration WithDeadline(DateTime deadline) => new Expiration(null, deadline);
+
+        /// <summary>
+        /// An <see cref="Expiration"/> with no timeout or deadline.
+        /// </summary>
+        /// <remarks>
+        /// Indicates that no expiration is required.
+        /// </remarks>
+        public static Expiration None { get; } = new Expiration(null, null);
+
+        private Expiration(TimeSpan? timeout, DateTime? deadline)
```

Comment:
Counterexample to "no constructors": everyone's used to calling `WebRequest.Create`